src/item.cpp

```diff
@@ -4393,6 +4393,27 @@ item *item::get_usable_item( const std::string &use_name )
     return nullptr;
 }
 
+long item::usages_remaining( const Character& ch ) const
+{
+    // items counted by charge always consume one charge per usage
+    if( count_by_charges() ) {
+        return charges;
+    }
+
+    // for items not consuming ammo then allow infinite usages
```

Comment:
Things like traps get used up instead. They consume no ammo, so `ammo_required()` should return 0.